uaclient/system.py

```diff
@@ -370,14 +370,26 @@ def create_file(filename: str, mode: int = 0o644) -> None:
     os.chmod(filename, mode)
 
 
-def write_file(filename: str, content: str, mode: int = 0o644) -> None:
+def write_file(
+    filename: str, content: str, mode: Optional[int] = None
+) -> None:
     """Write content to the provided filename encoding it if necessary.
 
     @param filename: The full path of the file to write.
     @param content: The content to write to the file.
     @param mode: The filesystem mode to set on the file.
     """
     tmpf = None
+    is_file_present = os.path.isfile(filename)
+    if is_file_present:
+        file_stat = pathlib.Path(filename).stat()
+        f_mode = util.get_file_perm_mode(filename)
+        if not mode:
+            mode = f_mode
+
+    else:
+        if not mode:
+            mode = 0o644
```

Comment:
Please double check that this is equivalent, but I think we can do this instead of `stat`-ing the file twice. Then we don't need the util function

```suggestion
    is_file_present = os.path.isfile(filename)
    if is_file_present:
        file_stat = pathlib.Path(filename).stat()
        f_mode = stat.S_IMODE(file_stat.st_mode)
        if mode is None:
            mode = f_mode
    elif mode is None:
        mode = 0o644
```